modules/gdscript/editor_templates/CharacterBody2D/basic_movement.gd

```diff
@@ -0,0 +1,29 @@
+# meta-description: Classic movement for gravity games (platformer, ...)
+# meta-default: true
+
+extends _BASE_
+
+const SPEED: int = 300
+const JUMP_FORCE: int = -400
+
+# Get the gravity from the project settings to be synced with RigidDynamicBody nodes.
+var gravity: int = ProjectSettings.get_setting("physics/2d/default_gravity")
+
+
+func _physics_process(delta):
+	# Add the gravity.
+	if not is_on_floor():
+		motion_velocity.y += gravity * delta
+
+	# Handle Jump.
+	if Input.is_action_just_pressed("ui_accept") and is_on_floor():
+		motion_velocity.y = JUMP_FORCE
+
+	#  Get the input direction and handle the movement/deceleration.
+	var direction = Input.get_axis("ui_left", "ui_right")
```

Comment:
I understand why it's done that way, but I don't really like that we encourage a bad practice by reusing **UI** actions for character movement. Those should only be used for UI purposes, character movement should be done with new actions customized for each game, such as `move_left` or `steer_left`, etc.

Of course we don't want to add such gameplay-specific actions to the default InputMap, so I don't have a good alternative to propose.

Reply:
What about adding a comment that explains what the good practice is (replace "ui_left" and "ui_right" with specific gameplay actions added to Project Settings > Input Mapping) so it still works out of the box but we encourage users to do things the proper way?